acct/check

```diff
@@ -132,11 +136,57 @@ if [ -f "$vhost_file" ] && vhost=$(grep -E "^[#[:space:]]*$user[![:space:]]" "$v
 fi
 
 # check apphosting config
-if [ -f "$vhost_app_file" ] && vhost=$(grep -E "^[#[:space:]]*$user[![:space:]]" "$vhost_app_file"); then
+if [ -f "$vhost_app_file" ] && vhost_app=$(grep -E "^[#[:space:]]*$user[![:space:]]" "$vhost_app_file"); then
   echo
   echo "Virtual apphosts:"
   echo "-------------------------------------------------------------------------------"
-  echo "$vhost"
+  echo "$vhost_app"
+  echo "-------------------------------------------------------------------------------"
+fi
+
+# check virtual mail config
+if [ -f "$vhost_mail_file" ] && vhost_mail=$(grep -E "^[#[:space:]]*$user[![:space:]]" "$vhost_mail_file"); then
+  echo
+  echo "Virtual mail:"
+  echo "-------------------------------------------------------------------------------"
+  echo "$vhost_mail"
+  echo "-------------------------------------------------------------------------------"
+fi
+
+
+# check dns entries
+expand_fqdns()
+{
+  while read -r entry; do
+    if [ -z "$entry" ]; then
+      continue
+    fi
+    
+    username=$(echo "$entry" | cut -d' ' -f1)
+    domain=$(echo "$entry" | cut -d' ' -f2)
+
+    # shellcheck disable=SC2143,SC2086,SC2063
+    if [ "$domain" = "-" ]; then
+      echo "$username.berkeley.edu"
+    elif [ "$(echo $domain | grep -E '*\.[a-z]+$')" ]; then
+      echo "$domain"
+    else
+      echo "$domain.berkeley.edu"
+    fi
+  done
+}
+
+fqdns=$(printf "%s\n%s\n%s" "$vhost" "$vhost_app" "$vhost_mail" | grep -v '#' | expand_fqdns)
```

Comment:
You need to add a final `\n` to the `printf` format string, or else `expand_fqdns` is going to miss the last line in the `vhost_mail` entries

Reply:
nice